### DISCUSSION TITLE
Data object metadata

### DISCUSSION THREAD

**Discussion by ll4strw** [UtrechtUniversity/yoda#347] [Q&A] [answered]
Hello Yoda team, 
let us suppose I have custom rules that I have added to the default yoda ruleset to add specific metadata upon data upload to the portal.
1. Is there a way to visualize in the yoda GUI all metadata associated to a given data object?
2. If a data object with added metadata is part of a data package submitted to the vault, the added data-object metadata `disappears` during the move from pending-deposit to vault. Is this the intended behavior? 

PS: I am testing the dev docker version of yoda.

**Comment by ll4strw** [comment]
To elaborate more on point 2:

```
# data-deposit stage
[irods@provider ~]$ imeta ls -d test
AVUs defined for dataObj /tempZone/home/deposit-pilot/deposit-pilot[1706795530]/test:
attribute: country
value: NL
units: 
----
attribute: org_replication_scheduled
value: irodsResc,irodsRescRepl,61
units: 

```

then 

```
# vault stage
[irods@provider ~]$ imeta ls -d  /tempZone/home/vault-pilot/deposit-pilot[1706795530][1706795726]/original/test
AVUs defined for dataObj /tempZone/home/vault-pilot/deposit-pilot[1706795530][1706795726]/original/test:
attribute: org_replication_scheduled
value: irodsResc,irodsRescRepl,44
units: 

```

**Reply by ll4strw** [reply]
I see there is an issue registered about this #19. Has this been implemented? Preserving data objects metadata is a very important issue.

**Reply by lwesterhof** [reply]
1. Only metadata on data package level is visualized in the web portal.

2. Copying file metadata to the vault has not (yet) been implemented and is currently not planned. 

You are welcome to open feature requests describing your use case. Thanks!

**Reply by ll4strw** [reply, marked as answer]
Hi @lwesterhof thanks for your message. I see there is already a feature request for data objects metadata:  #19 . Perhaps can that be re-open? Similarly, #344 requests that data objects metadata could be visualized/exported.
My use-case scenario is simple. I have custom iRODS rules that, upon file upload, automatically extract the file metadata (if any) to store them into iRODS. Think of image files with metadata in Exif format or a simple jupyter notebook  with arbitrary metadata as specified in the Jupyter Notebook Format, etc...
Researchers need to access and search such metadata so it would be great if they were preserved. 
A possible solution would be to include them both in json files and the iCAT, but I have no special requests on this as long as the metadata are preserved in iRODS. Thanks.
